cql3/restrictions/single_column_primary_key_restrictions.hh

```diff
@@ -55,6 +57,14 @@ namespace cql3 {
 
 namespace restrictions {
 
+inline
+void check_cartesian_product_size(size_t size, size_t max) {
+    if (size > max) {
+        throw std::runtime_error(fmt::format("Primary key cartesian product size {} is greater than maximum {}",
+                size, max));
```

Comment:
Wouldn't it be useful to know, which kind of key (partition or clustering) is this? Can we ever have both in a single query? (I think we can).

Reply:
Yes

Reply:
added